cmd/thanos/flags.go

```diff
@@ -73,3 +75,99 @@ func regSelectorRelabelFlags(cmd *kingpin.CmdClause) *extflag.PathOrContent {
+			"Time to wait after an interrupt received for GRPC Server.").
+			Default("2m")),
+		tlsSrvCert: *cmd.Flag("grpc-server-tls-cert",
+			"TLS Certificate for gRPC server, leave blank to disable TLS").
+			Default("").String(),
+		tlsSrvKey: *cmd.Flag("grpc-server-tls-key",
+			"TLS Key for the gRPC server, leave blank to disable TLS").
+			Default("").String(),
+		tlsSrvClientCA: *cmd.Flag("grpc-server-tls-client-ca",
+			"TLS CA to verify clients against. If no client CA is specified, there is no client verification on server side. (tls.NoClientCert)").
+			Default("").String(),
+	}
+}
+
+func parseFlagsToHttpConfig(cmd *kingpin.CmdClause) httpConfig {
+	return httpConfig{
+		bindAddress: *cmd.Flag("http-address",
+			"Listen host:port for HTTP endpoints.").
+			Default("0.0.0.0:10902").String(),
+		gracePeriod: *modelDuration(cmd.Flag("http-grace-period",
+			"Time to wait after an interrupt received for HTTP Server.").
+			Default("2m")),
+	}
+}
+
+func parseFlagsToPrometheusConfig(cmd *kingpin.CmdClause) prometheusConfig {
+	return prometheusConfig{
+		url: *cmd.Flag("prometheus.url",
+			"URL at which to reach Prometheus's API. For better performance use local network.").
+			Default("http://localhost:9090").URL(),
+		readyTimeout: *cmd.Flag("prometheus.ready_timeout",
+			"Maximum time to wait for the Prometheus instance to start up").
+			Default("10m").Duration(),
+	}
+}
+
+func parseFlagsToConnectionConfig(cmd *kingpin.CmdClause) connectionConfig {
+	return connectionConfig{
+		maxIdleConns: *cmd.Flag("receive.connection-pool-size",
+			"Controls the http MaxIdleConns. Default is 0, which is unlimited").
+			Int(),
+		maxIdleConnsPerHost: *cmd.Flag("receive.connection-pool-size-per-host",
+			"Controls the http MaxIdleConnsPerHost").
+			Default("100").Int(),
+	}
+}
+
+func parseFlagsToTsdbConfig(cmd *kingpin.CmdClause) tsdbConfig {
+	return tsdbConfig{
+		path: *cmd.Flag("tsdb.path",
+			"Data directory of TSDB.").
+			Default("./data").String(),
+	}
+}
+
+func parseFlagsToReloaderConfig(cmd *kingpin.CmdClause) reloaderConfig {
+	return reloaderConfig{
+		confFile: *cmd.Flag("reloader.config-file",
+			"Config file watched by the reloader.").
+			Default("").String(),
+		envVarConfFile: *cmd.Flag("reloader.config-envsubst-file",
+			"Output file for environment variable substituted config file.").
+			Default("").String(),
+		ruleDirectories: *cmd.Flag("reloader.rule-dir",
+			"Rule directories for the reloader to refresh (repeated field).").
+			Strings(),
+	}
+}
+
+func parseFlagsToShipperConf(cmd *kingpin.CmdClause) shipperConfig {
+	return shipperConfig{
+		uploadCompacted: *cmd.Flag("shipper.upload-compacted",
+			"If true sidecar will try to upload compacted blocks as well. Useful for migration purposes. Works only if compaction is disabled on Prometheus. Do it once and then disable the flag when done.").
+			Default("false").Bool(),
+		ignoreBlockSize: *cmd.Flag("shipper.ignore-unequal-block-size",
```

Comment:
So to make it work we need opposite: We need `BoolVar(&ignoreBlockSize)`.

How it works is that you register variables here first. Kingpin either returns address of the variable that the value will be AFTER parsing or it allows us to tell kingpin where is the address (using `*Var()` methods). Let's do the later so our config struct will fill (: